### PR TITLE
[TT-6835] Import via dashboard

### PR DESCRIPTION
**Pull request by marksou**

### CODE REVIEW

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md [hunk 1]
```diff
@@ -640,4 +640,131 @@ For more information on how Tyk builds the middleware operations structure in or
 ```
 #### What did you just do?
 
-You created an API that which tells your Tyk Gateway to validate any incoming request, against the JSON schema defined in the OAS API Definition.
\ No newline at end of file
+You created an API that which tells your Tyk Gateway to validate any incoming request, against the JSON schema defined in the OAS API Definition.
```
Comment (remotelycoding) [inline tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: Fixed some typos:

'You created an API which tells your Tyk Gateway to validate any incoming request against the JSON schema defined in the OAS API Definition.'

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md [hunk 2]
```diff
@@ -640,4 +640,131 @@ For more information on how Tyk builds the middleware operations structure in or
...
+
+- From an OpenAPI (JSON format) file
+- From an Existing Tyk API definition
+- From a SOAP WSDL definition file
+
+#### Importing an OpenAPI Definition
+
+Tyk supports import of version 2.0 and 3.0.x OpenAPI documents to create an API Definition.
+
+When importing OpenAPI documents using the Dashboard, Tyk detects the version of the document and generates either a native Tyk API Definition (for OpenAPI 2.0) or the new Tyk OAS API Definition (for OpenAPI 3.0.x).
```
Comment (remotelycoding) [inline tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: This isn't quite true. Updated wording:

'and generates... (for OpenAPI 3.0.x).'

should be


'and, by default, generates...(for OpenAPI 3.0.x). There is an option to create a Tyk Classic API Definition from an OpenAPI 3.0.x document if you wish.'

Also, uses of 'Tyk API Definition' referring to the non-OAS version, should be 'Tyk Classic API Definition'.

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md [hunk 3]
```diff
@@ -640,4 +640,131 @@ For more information on how Tyk builds the middleware operations structure in or
...
+#### Import an OpenAPI 3.0 Definition
+1. From the Import API screen, select OpenAPI.
+
+{{< img src="/img/oas/open-api-format.png" alt="Import OAS 3.0 API" >}}
+
+1. Paste your OpenAPI 3.0 definition into the code editor.
+
+{{< img src="/img/oas/oas-3-code.png" alt="OAS 3.0 definition in Editor" >}}
+
+3. Note that the Dashboard has detected that an OpenAPI 3.0 definition has been imported and you now have to select between various manual and automatic configuration options.
```
Comment (remotelycoding) [inline tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: 'and you now have to select between...'


should be

'and you may now select between...'

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md [hunk 4]
```diff
@@ -640,4 +640,131 @@ For more information on how Tyk builds the middleware operations structure in or
...
+{{< img src="/img/oas/oas-3-import-options.png" alt="OAS 3.0 configuration options" >}}
+
+#### Manual Configuration options
+
+- **Custom Listen Path**: A default listen path of of `/` is set if you don't configure this option
+- **Custom Upstream URL**: The first URL listed in your `servers` section is used if you don't configure this option
+
+#### Automatic Configuration options
+
+- **Generate Validate Request Middleware**: You can automatically validate paths that have the `requestBody` and `schema` fields configured. This allows your Tyk Gateway to validate your request payload agains the schema provided in your definition. See [Request Validation]({{< ref "/content/getting-started/key-concepts/request-validation.md" >}}) for more details.
```
Comment (remotelycoding) [inline tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: Typo: 'agains' should be 'against'

**Review by remotelycoding — CHANGES_REQUESTED** [review]
I've put a few inline comments into the change to get a few more updates done. I hope I've done that correctly. Please shout if I haven't.

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md (hunk 4 above)
Comment (marksou) [reply tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: Fixed in new commit.

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md (hunk 3 above)
Comment (marksou) [reply tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: Fixed in new commit.

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md (hunk 2 above)
Comment (marksou) [reply tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: Fixed in new commit.

File: tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md (hunk 1 above)
Comment (marksou) [reply tyk-docs/content/getting-started/using-oas-definitions/import-an-oas-api.md]: Fixed in new commit.

**Review by remotelycoding — APPROVED** [review]

**Pull request merged**